packages/core/src/components/sections/Breadcrumb/Breadcrumb.tsx

```diff
@@ -4,36 +4,25 @@ import Breadcrumb from 'src/components/ui/Breadcrumb'
 
 import Section from '../Section'
 
-import {
-  ServerCollectionPageQueryQuery,
-  ServerProductPageQueryQuery,
-} from '@generated/graphql'
 import styles from './section.module.scss'
-
-type BreadcrumbContext =
-  | ServerProductPageQueryQuery['product']
-  | ServerCollectionPageQueryQuery['collection']
+import { isPDP, isPLP, usePage } from 'src/sdk/overrides/PageProvider'
 
 interface BreadcrumbSectionProps {
-  context?: BreadcrumbContext
   icon: string
   alt: string
 }
 
-const isProduct = (x: any): x is ServerProductPageQueryQuery['product'] =>
-  x?.sku != undefined && x?.sku != null
-const isCollection = (
-  x: any
-): x is ServerCollectionPageQueryQuery['collection'] =>
-  x?.seo != undefined && x?.seo != null && x?.sku == undefined
-
-function BreadcrumbSection({ context, ...otherProps }: BreadcrumbSectionProps) {
-  const title = isCollection(context) ? context?.seo?.title : 'All Products'
+function BreadcrumbSection({ ...otherProps }: BreadcrumbSectionProps) {
+  const context = usePage()
+  const title = isPLP(context)
+    ? context?.data?.collection?.seo?.title
+    : 'All Products'
   const fallback = [{ item: '/', name: title, position: 1 }]
-  const breadcrumbList = isProduct(context)
-    ? context?.breadcrumbList?.itemListElement
-    : isCollection(context)
-    ? context?.breadcrumbList?.itemListElement
+
+  const breadcrumbList = isPDP(context)
+    ? context?.data?.product?.breadcrumbList?.itemListElement
+    : isPLP(context)
+    ? context?.data?.collection?.breadcrumbList?.itemListElement
     : fallback
```

Comment:
Do we really need know the type to return the correct `itemListElement` or we can do this suggestion?

```suggestion
  const breadcrumbList = 
    context?.data?.product?.breadcrumbList?.itemListElement ??
    context?.data?.collection?.breadcrumbList?.itemListElement ??
    fallback
```

Reply:
we need to know the types otherwise the typescript complains.